awx/main/models/notifications.py

```diff
@@ -383,12 +383,16 @@ def context(self, serialized_job):
         and a url to the job run."""
         job_context = {'host_status_counts': {}}
         summary = None
-        if hasattr(self, 'job_host_summaries'):
-            summary = self.job_host_summaries.first()
-        if summary:
-            from awx.api.serializers import JobHostSummarySerializer
-            summary_data = JobHostSummarySerializer(summary).to_representation(summary)
-            job_context['host_status_counts'] = summary_data
+        try:
+            has_event_property = any([f for f in self.event_class._meta.fields if f.name == 'event'])
+        except NotImplementedError:
+            has_event_property = False
+        if has_event_property:
+            qs = self.get_event_queryset()
+            if qs:
+                events = qs.only('event_data').filter(event='playbook_on_stats')
+                summary = events.first().get_host_status_counts()
```

Comment:
The `.first()` isn't guaranteed to return a result. We have existing race conditions with non-job type jobs where the `playbook_on_stats` event _may_ not exist. We should wrap this in another if statement i.e. `if events:`

Reply:
if we remove the `.first()`, I doubt it will even trigger the notification for Jobs/Project updates?  I've run few tests with the change you suggested but with those changes, I don't see the Tower sending notifications for Job and Project Updates. I may be missing something but I find the latest commit more reliable.

Reply:
Keep the call to `first()` but note that `first()` _could_ return `None`. Subsequently calling `.get_host_status_counts()` on `None` will raise an exception.